Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -26,128 +27,183 @@
-		} catch (UnknownHostException ex) {
-			if (log) PrefHelper.Debug(getClass().getSimpleName(), "Http connect exception: " + ex.getMessage());
-			return new ServerResponse(tag, NO_CONNECTIVITY_STATUS);
-		} catch (IOException ex) { 
-			if (log) PrefHelper.Debug(getClass().getSimpleName(), "IO exception: " + ex.getMessage());
-			return new ServerResponse(tag, 500);
-		}
-		return null;
-	}
-
-	public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout) {
-		return make_restful_post(body, url, tag, timeout, true, null);
-	}
-	
-	public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, BranchLinkData linkData) {
-		return make_restful_post(body, url, tag, timeout, true, linkData);
-	}
-
-	public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, boolean log) {
-		return make_restful_post(body, url, tag, timeout, log, null);
-	}
-	
-	public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, boolean log, BranchLinkData linkData) {
-		try {    	
-			if (body.has("app_id") && body.getString("app_id").equals(PrefHelper.NO_STRING_VALUE)) {
-				return new ServerResponse(tag, NO_API_KEY_STATUS);
-			}
-			
-			body.put("sdk", "android" + SDK_VERSION);
-			if (log) {
-				PrefHelper.Debug("BranchSDK", "posting to " + url);
-				PrefHelper.Debug("BranchSDK", "Post value = " + body.toString());
-			}
-		    HttpPost request = new HttpPost(url);
-		    request.setEntity(new ByteArrayEntity(body.toString().getBytes("UTF8")));
-		    request.setHeader("Content-type", "application/json");
-		    HttpResponse response = getGenericHttpClient(timeout).execute(request);
-		    return processEntityForJSON(response.getEntity(), response.getStatusLine().getStatusCode(), tag, log, linkData);
-
-		} catch (SocketException ex) {
-			if (log) PrefHelper.Debug(getClass().getSimpleName(), "Http connect exception: " + ex.getMessage());
-			return new ServerResponse(tag, NO_CONNECTIVITY_STATUS);
-		} catch (UnknownHostException ex) {
-			if (log) PrefHelper.Debug(getClass().getSimpleName(), "Http connect exception: " + ex.getMessage());
-			return new ServerResponse(tag, NO_CONNECTIVITY_STATUS);
-		} catch (Exception ex) {
-			if (log) PrefHelper.Debug(getClass().getSimpleName(), "Exception: " + ex.getMessage());
-			if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.HONEYCOMB) {
-				if (ex instanceof NetworkOnMainThreadException)
-					Log.i("BranchSDK", "Branch Error: Don't call our synchronous methods on the main thread!!!");
-			}
-			return new ServerResponse(tag, 500);
-		}
-	}
+    public static final String NO_TAG_VALUE = "no_tag";
+    public static final int NO_CONNECTIVITY_STATUS = -1009;
+    public static final int NO_API_KEY_STATUS = -1234;
+
+    private static final String SDK_VERSION = "1.4.4";
+    private static final int DEFAULT_TIMEOUT = 3000;
+
+    protected PrefHelper prefHelper_;
```

Comment:
Besides moving prefHelper_ here, can I assume all the other lines are for tab -> spaces changes? It's hard to compare so many lines one by one :-)